### DISCUSSION TITLE
Plugin rewrite

### DISCUSSION THREAD

**Discussion by reuseman** [reuseman/flashcards-obsidian#155] [General]
This is to discuss the plugin rewrite #151.

**Comment by bytewife** [comment]
I'll outline a few ideas below:

**Comment by bytewife** [comment]
- Modular separation between PKB & flashcard review. Structure the project like:
```
PKB (i.e. Obsidian) 
|
PKB-to-plugin bridge
|
plugin (flashcards-obsidian)
|
plugin-to-reviewer bridge 
|
flashcard reviewer (i.e. Anki)
```

**Reply by reuseman** [reply]
The high level overview of the architecture is totally sound

**Comment by bytewife** [comment]
Remnote-like WYSIWYG for UI:
- Hide the flashcard ID from the user in Live Preview mode
- Hide the plugin's syntax (i.e. `#card` tags, `{}` clozes, etc) from the user in Live Preview mode. Alternatively, change it to a cleaner symbol that can be clicked on, like in Remnote

**Reply by reuseman** [reply]
- What do you mean by hiding the flashcard ID in preview mode? It should not be visible at the moment?
- How would you hide the #card tag? There should be something that makes clear that a specific block is a card, especially because nowadays on Obsidian you have the live preview mode where you can even write. Thus, if you don't see the flashcard tag there it may be misleading

**Comment by bytewife** [comment]
A few other ideas, such as
- Cloze Lists (drawing inspiration from Andy Matuschak's [Prompts essay](https://andymatuschak.org/prompts/#:~:text=them%20more%20later.-,Lists,-Back%20to%20the))
- User-chosen syntax for clozes & other similar features
- A way of attaching tags to Anki cards, so that you can filter out different cards from a single Anki deck
- Export notes, removing plugin-specific syntax from notes
- Fine-tune control of how much context in prior and following lines of text to include in each Anki card (in particular with cloze and similar features)
- Search for "orphaned" flashcards (flashcards that exist in Obsidian, but not in Anki)

**Reply by reuseman** [reply]
1. Need to read more about prompts essay, seems interesting. What would be needed in order to have it, that it's not available right now?
2. It makes sense, but I would always keep in mind as a goal that the less customization, the fewer bugs. Especially in the initial rewrite, let's focus on a specific syntax in order to deliver asap, and then we can tweak the custom stuff.
3. Right now you can use a specific tag for each question, what do you mean?
4. This would be nice, but I guess it should be compatible with Obsidian export command, if that's possible.
5. Like point number 2. It's ok, but let's focus on a quick deliver of the initial part of the rewrite.
6. I think it could be useful with the current plugin, but if you have a command that generates globally the flashcards, this would probably never be used.
